rcgen/src/error.rs

```diff
@@ -98,6 +101,36 @@ impl fmt::Display for Error {
 
 impl std::error::Error for Error {}
 
+/// Invalid ASN.1 string type
+#[derive(Debug, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum InvalidAsn1String {
+	/// Invalid PrintableString type
+	PrintableString(String),
+	/// Invalide UniversalString type
+	UniversalString(String),
+	/// Invalid Ia5String type
+	Ia5String(String),
+	/// Invalid TeletexString type
+	TeletexString(String),
+	/// Invalid BmpString type
+	BmpString(String),
+}
+
+impl fmt::Display for InvalidAsn1String {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		use InvalidAsn1String as E;
```

Comment:
Let's just do `use InvalidAsn1String::*;` here.

Reply:
Using the variants name with a wildcard can lead to difficult diag bug. If we remove a variant of the `enum`, then in the match the old variant become a variable that will be a catch-all pattern. The warning that will be emitted will be  a "non snake case name" that's difficult to diag.
This [video](https://youtu.be/8j_FbjiowvE?feature=shared&t=97) explains this better than I do.

I wanted to open a PR after this one to change the code when we use this pattern.

Reply:
The only lint I found is [one which prevent this pattern for stylistic purpose](https://rust-lang.github.io/rust-clippy/master/index.html#/enum_glob_use). Please do you have a link?

I don't feel strongly about it and I can use the variant with a wildcard as the rest of the crate does. If we want to tackle this one day, maybe it's better to do all this in one separate PR